### PR TITLE
[WIP]

### PR DESCRIPTION
**Pull request by sukeshni**

### CODE REVIEW

**Comment by sukeshni** [comment]
@shunjikonishi Please review

File: src/main/scala/codecheck/github/operations/CollaboratorOp.scala [hunk 1]
```diff
@@ -20,4 +20,28 @@ trait CollaboratorOp {
       }
     )
   }
+
+  def isCollaborator(owner: String, repo: String, name: String): Future[Boolean] = {
+    val path = s"/repos/${owner}/${repo}/collaborators/" + encode(name)
+    exec("GET", path).map { res =>
```
Comment (shunjikonishi) [inline src/main/scala/codecheck/github/operations/CollaboratorOp.scala]: You have to pass fail404=false

File: src/test/scala/CollaboratorOpSpec.scala [hunk 2]
```diff
@@ -16,4 +18,43 @@ class CollaboratorOpSpec extends FunSpec with Constants {
       assert(c.site_admin == false)
     }
   }
+  describe("isCollaborator"){
+    it("if it is Collaborator"){
+      val res = Await.result(api.isCollaborator(organization, repo, user),TIMEOUT)
```
Comment (shunjikonishi) [inline src/test/scala/CollaboratorOpSpec.scala]: I think it is better to use user repo.

File: src/test/scala/CollaboratorOpSpec.scala [hunk 3]
```diff
@@ -16,4 +18,43 @@ class CollaboratorOpSpec extends FunSpec with Constants {
...
+      assert(res == true)
+    }
+    it("if it is not a Collaborator"){
+      val res1 = Await.result(api.isCollaborator(organization, repo, otherUserInvalid).failed,TIMEOUT)
+      res1 match {
+        case e: NotFoundException =>
```
Comment (shunjikonishi) [inline src/test/scala/CollaboratorOpSpec.scala]: It should return false.(use fail404)

File: src/test/scala/Constants.scala [hunk 4]
```diff
@@ -12,7 +12,10 @@ trait Constants {
   protected val TIMEOUT = 5 seconds
   protected val api = Constants.API
 
-  protected val user = "fanwashere" //REQUIRED: Edit this to your own username.
+  protected val user = "sukeshni" //REQUIRED: Edit this to your own username.
```
Comment (shunjikonishi) [inline src/test/scala/Constants.scala:15]: FYI @fanwashere

It is not good to rewrite this line by each developer.
Use sys.env("xxxxx")

And write required environment variables to README.

**Comment by shunjikonishi** [comment]
@sukeshni fix me

About Constants and environment variables, I'll make another issue.
So, you don't have to fix it in this branch.

FYI: @fanwashere

**Comment by shunjikonishi** [comment]
@sukeshni 

When you make pull request, please name a simple title.
Just '[WIP]' is confusing.

**Comment by sukeshni** [comment]
@shunjikonishi will fix it

**Comment by sukeshni** [comment]
@fanwashere Review me

**Comment by fanwashere** [comment]
LGTM

@sukeshni Tag me to review again once you are done with Constants.scala.

**Comment by sukeshni** [comment]
@fanwashere 
Please review https://github.com/code-check/github-api-scala/pull/33 for Constants.scala.

**Comment by sukeshni** [comment]
@shunjikonishi Please review

**Comment by shunjikonishi** [comment]
@sukeshni I've checked it with #33 

I guess you made 30-UserEnv branch from this branch.
It is better to make branch always from master.

``` bash
git co -b new-branch master
```

#33 includes all commits of this pullrequest.
So, you don't have to merge this.

**Comment by sukeshni** [comment]
@shunjikonishi yes you are right it is better ti make branch from master, my mistake.
Won't merge this branch.

**Comment by sukeshni** [comment]
@shunjikonishi Oh it merged when i merged #33.
Will delete this branch.

**Pull request merged**